src/query.rs

```diff
@@ -2,7 +2,39 @@
 
 use std::convert::From;
 use linear_map::LinearMap;
-use value::*;
+use value::{Key, Pointer, Value};
+
+/// Specifies a positive numeric range of data.
+pub struct Range(Bound<u64>, Bound<u64>);
```

Comment:
Can't have unbounded ranges with rust's `Range`. In the future I believe rust will have a `Range<T>` type using their `Bound`s, so at that time we can refactor.

For the meantime I'll turn `Bound<T>` into `Bound`, replacing the `T`s with `u64`.